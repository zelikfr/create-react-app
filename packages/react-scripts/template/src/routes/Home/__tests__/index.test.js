import React from 'react'
import { mount } from 'enzyme'
import HomePage from '../'

describe('HomePage', () => {
  it('should create a component wrapped with HomePage', () => {
    // When I render the HomePage element
    const component = mount(<HomePage call={jest.fn((opts, succ) => succ({ data: [] }))} />)
    // Then I should have the component to match the snapshot.
    expect(component).toMatchSnapshot()
  })
  it('should create a component wrapped with HomePage and data in call', () => {
    // When I render the HomePage element
    const component = mount(<HomePage call={jest.fn((opts, succ) => succ({ data: [{ id: 1, 'first_name': 'John' }] }))} />)
    // Then I should have the component to match the snapshot.
    expect(component).toMatchSnapshot()
  })
  it('should logout after click on button', () => {
    // When I render the HomePage element
    const logout = jest.fn()
    const component = mount(<HomePage logout={logout} call={jest.fn((opts, succ) => succ({ data: [{ id: 1, 'first_name': 'John' }] }))} />)
    component.find('Button').simulate('click')
    // Then I should have the component to match the snapshot.
    expect(logout).toHaveBeenCalled()
  })
})
