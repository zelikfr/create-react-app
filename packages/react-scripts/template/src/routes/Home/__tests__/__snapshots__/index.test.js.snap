// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should create a component wrapped with HomePage 1`] = `
<WithAPI
  call={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "method": "get",
            "url": "users",
          },
          [Function],
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <Home
    call={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "method": "get",
              "url": "users",
            },
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  >
    <div>
      <h1>
        Hello world
      </h1>
      <div>
        <ul />
      </div>
      <Button
        onClick={[Function]}
      >
        <button
          className="button"
          onClick={[Function]}
        >
          Logout
        </button>
      </Button>
    </div>
  </Home>
</WithAPI>
`;

exports[`HomePage should create a component wrapped with HomePage and data in call 1`] = `
<WithAPI
  call={
    [MockFunction] {
      "calls": Array [
        Array [
          Object {
            "method": "get",
            "url": "users",
          },
          [Function],
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <Home
    call={
      [MockFunction] {
        "calls": Array [
          Array [
            Object {
              "method": "get",
              "url": "users",
            },
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  >
    <div>
      <h1>
        Hello world
      </h1>
      <div>
        <ul>
          <li
            key="1"
          >
            John
          </li>
        </ul>
      </div>
      <Button
        onClick={[Function]}
      >
        <button
          className="button"
          onClick={[Function]}
        >
          Logout
        </button>
      </Button>
    </div>
  </Home>
</WithAPI>
`;
